siad/daemon.go

```diff
@@ -125,6 +139,7 @@ func startDaemon(config Config) (err error) {
 		if err != nil {
 			return err
 		}
+		defer g.Close()
```

Comment:
What was your reason for using multiple `defer`s instead of one `defer srv.Close()`?